### ISSUE TITLE
OpenAPI $ref loop when setting Schema implementation

### ISSUE THREAD

**Issue by oaklandcorp-jkaiser** [quarkusio/quarkus#34233]
### Describe the bug

Last year I ran into an issue trying to represent a file upload cleanly in OpenAPI and the Swagger UI. I created a [StackOverflow post](https://stackoverflow.com/questions/74465077/how-to-view-a-fileupload-response-body-as-an-upload-button-when-seen-through-swa) asking how this could be done using the approach suggested by the RESTEasy Reactive Guide. An answer was posted referencing [this Quarkus issue](https://github.com/quarkusio/quarkus/issues/28231#issuecomment-1259848398) that seems to provide a fairly straight forward way to correct the format of the file upload field. This suggestion has served me well for the life of Quarkus 2, but now that I'm trying to migrate to Quarkus 3 it no longer works. In fact it's worse because it seems to create a `$ref` loop in the generated OpenAPI document.

### Expected behavior

Generated model in OpenAPI document:
```
UploadItemSchema:
  format: binary
  type: string
```

### Actual behavior

Generated model in OpenAPI document:
```
UploadItemSchema:
  $ref: '#/components/schemas/UploadItemSchema'
````

### How to Reproduce?

The [StackOverflow post](https://stackoverflow.com/questions/74465077/how-to-view-a-fileupload-response-body-as-an-upload-button-when-seen-through-swa) has small example of the solution that still closely reflects that app that I'm migrating.

### Output of `uname -a` or `ver`

Linux tester 5.19.0-45-generic #46~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Jun 7 15:06:04 UTC 20 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.6" 2023-01-17 LTS OpenJDK Runtime Environment Corretto-17.0.6.10.1 (build 17.0.6+10-LTS) OpenJDK 64-Bit Server VM Corretto-17.0.6.10.1 (build 17.0.6+10-LTS, mixed mode, sharing)

### Quarkus version or git rev

3.1.2.Final

**Comment by MikeEdgar** [comment]
This should be fixed by #34174

**Comment by oaklandcorp-jkaiser** [comment]
Are there any other workarounds to consider that would allow a `FileUpload` body to have the correct OpenAPI metadata set before that fix is available? Just curious in case that fix isn't available until 3.3.
